src/coreclr/src/jit/lowerarmarch.cpp

```diff
@@ -706,12 +706,14 @@ void Lowering::LowerHWIntrinsicCmpOp(GenTreeHWIntrinsic* node, genTreeOps cmpOp)
 
     if ((baseType == TYP_FLOAT) && (simdSize == 12))
     {
-        // For TYP_SIMD12 we need to clear the upper bits and can't assume their value
+        // For TYP_SIMD12 we don't want the upper bits to participate in the comparison. So, we will insert all ones
+        // into those bits of the result, "as if" the upper bits are equal. Then if all lower bits are equal, we get the 
+        // expected all-ones result, and will get the expected 0's only where there are non-matching bits.
```

Comment:
Thanks for making the comment change - it probably overflowed the allowed line length causing the formatting error? Sorry about that!